### PR TITLE
Create developer documentation describing the format of a SNS (success/failure) message

### PR DESCRIPTION
**Pull request by flamingbear**
**Summary:** Create developer documentation describing the format of a SNS (success/failure) message

Addresses [CUMULUS-723: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-723)

## Changes
* Adds section on SNS message format to the sns.md portion of the data-cookbook

CUMULUS-723 had acceptance criteria `Developer documentation (location TBD) exists with the format of the SNS failure and success message`

but looking at the [parent requirement CUMULUS-634](https://bugs.earthdata.nasa.gov/browse/CUMULUS-634) it seems that these requirements
`Developer documentation describes the format of the success message`
`Developer documentation describes the format of the failure message` are what we need.

I think this PR describes what a Developer would need in order to subscribe to a the sftracker SNS Topic and understand what was included in the message.

### CODE REVIEW

File: docs/data-cookbooks/cnm-workflow.md [hunk 1]
```diff
@@ -568,7 +568,7 @@ When the `kinesisConsumer` fails to process a record, the failure is captured an
...
 
-This system ensures when `kinesisCosumer` fails to process a record and trigger a workflow, the record is retried 3 times. This retry behavior improves system reliability in case of any external service failure outside of Cumulus control. 
+This system ensures when `kinesisCosumer` fails to process a record and trigger a workflow, the record is retried 3 times. This retry behavior improves system reliability in case of any external service failure outside of Cumulus control.
 
 The Kinesis error handling system - the `kinesisFallback` SNS topic, `kinesisConsumer` lambda, and `kinesisFailure` SQS queue - come with the API package and do not need to be configured by the operator.
```
Comment (flamingbear) [inline docs/data-cookbooks/cnm-workflow.md]: Yeah, no idea what's up with that red stuff, but that's just a trailing whitespace deleted on L571.

File: docs/data-cookbooks/sns.md [hunk 2]
```diff
@@ -91,32 +91,42 @@ PdrStatusReport:
...
+
+### SNS message format
+
+The configured `SfSnsReport` lambda receives the Cumulus message [(as the lambda's task input)](../workflows/input_output.html#2-resolve-task-input) and is responsible for publishing the message to the sftracker SNS Topic. But before it publishes the message, `SfSnsReport` makes a determiniation about the workflow status and adds an additional metadata key to the message at `message.meta.status`.
+
+First it determines whether the workflow has finished by looking for the `sfnEnd` key in the `config` object.  If the workflow has finished it checks to see if it has failed by searching the input message for a non-empty `exception` object. The lambda updates the `message.meta.status` with `failed` or `completed` based on that result.  If the workflow is not finished the lambda sets `message.meta.status` to `running`.
+
+This means that subscribers to the sftracker SNS Topic can expect to find the published message by parsing the json string representation of the message found in the [SNS event](https://docs.aws.amazon.com/lambda/latest/dg/eventsources.html#eventsources-sns) at `Records[].Sns.Message` and examining the `meta.status` value.
```
Comment (laurenfrederick) [inline docs/data-cookbooks/sns.md]: Capitalize JSON

Comment (laurenfrederick) [reply docs/data-cookbooks/sns.md]: Is the `Message` the full Cumulus message from the output of the workflow?

File: docs/data-cookbooks/sns.md [hunk 3]
```diff
@@ -91,32 +91,42 @@ PdrStatusReport:
...
 ```
 
-Make sure that your listener lambda is configured in `lambdas.yml`.
+Make sure that the receiver lambda is configured in `lambdas.yml`.
+
+### SNS message format
+
+The configured `SfSnsReport` lambda receives the Cumulus message [(as the lambda's task input)](../workflows/input_output.html#2-resolve-task-input) and is responsible for publishing the message to the sftracker SNS Topic. But before it publishes the message, `SfSnsReport` makes a determiniation about the workflow status and adds an additional metadata key to the message at `message.meta.status`.
+
+First it determines whether the workflow has finished by looking for the `sfnEnd` key in the `config` object.  If the workflow has finished it checks to see if it has failed by searching the input message for a non-empty `exception` object. The lambda updates the `message.meta.status` with `failed` or `completed` based on that result.  If the workflow is not finished the lambda sets `message.meta.status` to `running`.
```
Comment (laurenfrederick) [inline docs/data-cookbooks/sns.md]: I feel like there is a comma after 'If the workflow has finished' but maybe I'm wrong

File: docs/data-cookbooks/sns.md (hunk 2 above)
Comment (flamingbear) [reply docs/data-cookbooks/sns.md]: Records[0].Sns.Message = stringified object of the original full cumulus message + meta.status addition..

File: docs/data-cookbooks/sns.md (hunk 3 above)
Comment (flamingbear) [reply docs/data-cookbooks/sns.md]: It's imaginary.

**Pull request merged**